src/hooks/useTodoList.js

```diff
@@ -0,0 +1,171 @@
+import { useState, useEffect } from 'react';
+
+export function useTodoList() {
+    const [inputValue, setInputValue] = useState('');
+    const [activeItems, setActiveItems] = useState([]);
+    const [completedItems, setCompletedItems] = useState([]);
+    const [selectedItems, setSelectedItems] = useState([]);
+    const [isSelectingTodoList, setIsSelectingTodoList] = useState(false);
+    const [isSelectingDone, setIsSelectingDone] = useState(false);
+
+    // 초기 데이터 로드
+    useEffect(() => {
+        const savedActiveItems = JSON.parse(localStorage.getItem('activeItems')) || [];
+        const savedCompletedItems = JSON.parse(localStorage.getItem('completedItems')) || [];
+        setActiveItems(savedActiveItems);
+        setCompletedItems(savedCompletedItems);
+    }, []);
+
+    // localStorage 업데이트 함수
+    const updateLocalStorage = (activeItems, completedItems) => {
+        localStorage.setItem('activeItems', JSON.stringify(activeItems));
+        localStorage.setItem('completedItems', JSON.stringify(completedItems));
+    };
+
+    const addItem = () => {
+        if (inputValue.trim()) {
+            const newActiveItems = [...activeItems, inputValue];
+            setActiveItems(newActiveItems);
+            updateLocalStorage(newActiveItems, completedItems);
+            setInputValue('');
+        }
+    };
+
+    const toggleSelectingTodoList = () => {
+        setIsSelectingTodoList(!isSelectingTodoList);
+        setSelectedItems([]);
+
+    };
+
+    const toggleSelectingDone = () => {
+        setIsSelectingDone(!isSelectingDone);
+        setSelectedItems([]);
+    };
+
+    const toggleSelectItem = (index, isCompleted) => {
+        const items = isCompleted ? completedItems : activeItems;
+        if (selectedItems.includes(index)) {
+            setSelectedItems(selectedItems.filter((i) => i !== index));
+        } else {
+            setSelectedItems([...selectedItems, index]);
```

Comment:
`spread` 연산자 사용 아주 좋습니다! 다만 `setState`를 사용할 때 변경되는 state를 직접 참조하여 사용하는 것보다는 콜백 함수를 작성하는 방식이 조금 더 안정적인 방식입니다 🙂

```suggestion
            setSelectedItems(prev => [...prev, index]);
```

---

_그 이유가 궁금하다면..._

프로젝트를 진행하다 보면 `state 변경 -> 렌더링 -> state 변경 -> 렌더링 -> ...` 과정을 반복하게 됩니다.

`setState(state + 1)`은 현재의 state 값을 기반으로 상태를 업데이트 합니다. 하지만 이 방식은 state 값이 변경되는 과정에서 다른 setState 호출이 있을 경우, 이전 렌더링 시점의 상태를 기반으로 하기 때문에 최신 상태값을 반영하지 못할 수 있습니다.

반면 `setState(prev => prev + 1)`은 콜백 함수를 사용하여 이전 상태(prev)를 기준으로 상태를 업데이트 합니다. 이 방법은 상태 업데이트가 비동기적으로 처리되는 동안에도 항상 최신 상태를 반영하므로, 상태가 여러 번 업데이트 될 때에도 정확한 결과를 보장한다고 합니다!